drivers/include/mtd.h

```diff
@@ -199,6 +273,29 @@ int mtd_read(mtd_dev_t *mtd, void *dest, uint32_t addr, uint32_t count);
  */
 int mtd_write(mtd_dev_t *mtd, const void *src, uint32_t addr, uint32_t count);
 
+/**
+ * @brief   Write data to a MTD device with pagewise addressing
+ *
+ * @p offset + @p count must be inside a page boundary.
+ * Some devices might not support an @p offset != 0.
+ *
+ * @param      mtd   the device to write to
+ * @param[in]  src   the buffer to write
+ * @param[in]  page  the start page to write to
+ * @param[in]  offset byte offset from the start of the page
+ * @param[in]  count the number of bytes to write
```

Comment:
Can you give it a more speaking name? E.g. `size` or `n_bytes`? With `count` it is not immediately obvious that it is the number of bytes, not the number of sectors. (Sure, number of sectors makes no sense with an `offset` parameter there as well, but that takes ~ 1 second to realize.)

Some for the other APIs where a size in bytes is given.